// @aws-sdk/client-eventbridge/dist-es/models/models_0.js
import { __assign, __extends } from "tslib";
import { EventBridgeServiceException as __BaseException } from "./EventBridgeServiceException";
export var ActivateEventSourceRequest;
(function (ActivateEventSourceRequest) {
    ActivateEventSourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ActivateEventSourceRequest || (ActivateEventSourceRequest = {}));
var ConcurrentModificationException = (function (_super) {
    __extends(ConcurrentModificationException, _super);
    function ConcurrentModificationException(opts) {
        var _this = _super.call(this, __assign({ name: "ConcurrentModificationException", $fault: "client" }, opts)) || this;
        _this.name = "ConcurrentModificationException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ConcurrentModificationException.prototype);
        return _this;
    }
    return ConcurrentModificationException;
}(__BaseException));
export { ConcurrentModificationException };
var InternalException = (function (_super) {
    __extends(InternalException, _super);
    function InternalException(opts) {
        var _this = _super.call(this, __assign({ name: "InternalException", $fault: "server" }, opts)) || this;
        _this.name = "InternalException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, InternalException.prototype);
        return _this;
    }
    return InternalException;
}(__BaseException));
export { InternalException };
var InvalidStateException = (function (_super) {
    __extends(InvalidStateException, _super);
    function InvalidStateException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidStateException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidStateException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidStateException.prototype);
        return _this;
    }
    return InvalidStateException;
}(__BaseException));
export { InvalidStateException };
var OperationDisabledException = (function (_super) {
    __extends(OperationDisabledException, _super);
    function OperationDisabledException(opts) {
        var _this = _super.call(this, __assign({ name: "OperationDisabledException", $fault: "client" }, opts)) || this;
        _this.name = "OperationDisabledException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, OperationDisabledException.prototype);
        return _this;
    }
    return OperationDisabledException;
}(__BaseException));
export { OperationDisabledException };
var ResourceNotFoundException = (function (_super) {
    __extends(ResourceNotFoundException, _super);
    function ResourceNotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "ResourceNotFoundException", $fault: "client" }, opts)) || this;
        _this.name = "ResourceNotFoundException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ResourceNotFoundException.prototype);
        return _this;
    }
    return ResourceNotFoundException;
}(__BaseException));
export { ResourceNotFoundException };
export var ApiDestinationState;
(function (ApiDestinationState) {
    ApiDestinationState["ACTIVE"] = "ACTIVE";
    ApiDestinationState["INACTIVE"] = "INACTIVE";
})(ApiDestinationState || (ApiDestinationState = {}));
export var ApiDestinationHttpMethod;
(function (ApiDestinationHttpMethod) {
    ApiDestinationHttpMethod["DELETE"] = "DELETE";
    ApiDestinationHttpMethod["GET"] = "GET";
    ApiDestinationHttpMethod["HEAD"] = "HEAD";
    ApiDestinationHttpMethod["OPTIONS"] = "OPTIONS";
    ApiDestinationHttpMethod["PATCH"] = "PATCH";
    ApiDestinationHttpMethod["POST"] = "POST";
    ApiDestinationHttpMethod["PUT"] = "PUT";
})(ApiDestinationHttpMethod || (ApiDestinationHttpMethod = {}));
export var ApiDestination;
(function (ApiDestination) {
    ApiDestination.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ApiDestination || (ApiDestination = {}));
export var ArchiveState;
(function (ArchiveState) {
    ArchiveState["CREATE_FAILED"] = "CREATE_FAILED";
    ArchiveState["CREATING"] = "CREATING";
    ArchiveState["DISABLED"] = "DISABLED";
    ArchiveState["ENABLED"] = "ENABLED";
    ArchiveState["UPDATE_FAILED"] = "UPDATE_FAILED";
    ArchiveState["UPDATING"] = "UPDATING";
})(ArchiveState || (ArchiveState = {}));
export var Archive;
(function (Archive) {
    Archive.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Archive || (Archive = {}));
export var AssignPublicIp;
(function (AssignPublicIp) {
    AssignPublicIp["DISABLED"] = "DISABLED";
    AssignPublicIp["ENABLED"] = "ENABLED";
})(AssignPublicIp || (AssignPublicIp = {}));
export var CancelReplayRequest;
(function (CancelReplayRequest) {
    CancelReplayRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CancelReplayRequest || (CancelReplayRequest = {}));
export var ReplayState;
(function (ReplayState) {
    ReplayState["CANCELLED"] = "CANCELLED";
    ReplayState["CANCELLING"] = "CANCELLING";
    ReplayState["COMPLETED"] = "COMPLETED";
    ReplayState["FAILED"] = "FAILED";
    ReplayState["RUNNING"] = "RUNNING";
    ReplayState["STARTING"] = "STARTING";
})(ReplayState || (ReplayState = {}));
export var CancelReplayResponse;
(function (CancelReplayResponse) {
    CancelReplayResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CancelReplayResponse || (CancelReplayResponse = {}));
var IllegalStatusException = (function (_super) {
    __extends(IllegalStatusException, _super);
    function IllegalStatusException(opts) {
        var _this = _super.call(this, __assign({ name: "IllegalStatusException", $fault: "client" }, opts)) || this;
        _this.name = "IllegalStatusException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, IllegalStatusException.prototype);
        return _this;
    }
    return IllegalStatusException;
}(__BaseException));
export { IllegalStatusException };
export var CreateApiDestinationRequest;
(function (CreateApiDestinationRequest) {
    CreateApiDestinationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateApiDestinationRequest || (CreateApiDestinationRequest = {}));
export var CreateApiDestinationResponse;
(function (CreateApiDestinationResponse) {
    CreateApiDestinationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateApiDestinationResponse || (CreateApiDestinationResponse = {}));
var LimitExceededException = (function (_super) {
    __extends(LimitExceededException, _super);
    function LimitExceededException(opts) {
        var _this = _super.call(this, __assign({ name: "LimitExceededException", $fault: "client" }, opts)) || this;
        _this.name = "LimitExceededException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, LimitExceededException.prototype);
        return _this;
    }
    return LimitExceededException;
}(__BaseException));
export { LimitExceededException };
var ResourceAlreadyExistsException = (function (_super) {
    __extends(ResourceAlreadyExistsException, _super);
    function ResourceAlreadyExistsException(opts) {
        var _this = _super.call(this, __assign({ name: "ResourceAlreadyExistsException", $fault: "client" }, opts)) || this;
        _this.name = "ResourceAlreadyExistsException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ResourceAlreadyExistsException.prototype);
        return _this;
    }
    return ResourceAlreadyExistsException;
}(__BaseException));
export { ResourceAlreadyExistsException };
export var CreateArchiveRequest;
(function (CreateArchiveRequest) {
    CreateArchiveRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateArchiveRequest || (CreateArchiveRequest = {}));
export var CreateArchiveResponse;
(function (CreateArchiveResponse) {
    CreateArchiveResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateArchiveResponse || (CreateArchiveResponse = {}));
var InvalidEventPatternException = (function (_super) {
    __extends(InvalidEventPatternException, _super);
    function InvalidEventPatternException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidEventPatternException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidEventPatternException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidEventPatternException.prototype);
        return _this;
    }
    return InvalidEventPatternException;
}(__BaseException));
export { InvalidEventPatternException };
export var ConnectionAuthorizationType;
(function (ConnectionAuthorizationType) {
    ConnectionAuthorizationType["API_KEY"] = "API_KEY";
    ConnectionAuthorizationType["BASIC"] = "BASIC";
    ConnectionAuthorizationType["OAUTH_CLIENT_CREDENTIALS"] = "OAUTH_CLIENT_CREDENTIALS";
})(ConnectionAuthorizationType || (ConnectionAuthorizationType = {}));
export var CreateConnectionApiKeyAuthRequestParameters;
(function (CreateConnectionApiKeyAuthRequestParameters) {
    CreateConnectionApiKeyAuthRequestParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateConnectionApiKeyAuthRequestParameters || (CreateConnectionApiKeyAuthRequestParameters = {}));
export var CreateConnectionBasicAuthRequestParameters;
(function (CreateConnectionBasicAuthRequestParameters) {
    CreateConnectionBasicAuthRequestParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateConnectionBasicAuthRequestParameters || (CreateConnectionBasicAuthRequestParameters = {}));
export var ConnectionBodyParameter;
(function (ConnectionBodyParameter) {
    ConnectionBodyParameter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectionBodyParameter || (ConnectionBodyParameter = {}));
export var ConnectionHeaderParameter;
(function (ConnectionHeaderParameter) {
    ConnectionHeaderParameter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectionHeaderParameter || (ConnectionHeaderParameter = {}));
export var ConnectionQueryStringParameter;
(function (ConnectionQueryStringParameter) {
    ConnectionQueryStringParameter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectionQueryStringParameter || (ConnectionQueryStringParameter = {}));
export var ConnectionHttpParameters;
(function (ConnectionHttpParameters) {
    ConnectionHttpParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectionHttpParameters || (ConnectionHttpParameters = {}));
export var CreateConnectionOAuthClientRequestParameters;
(function (CreateConnectionOAuthClientRequestParameters) {
    CreateConnectionOAuthClientRequestParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateConnectionOAuthClientRequestParameters || (CreateConnectionOAuthClientRequestParameters = {}));
export var ConnectionOAuthHttpMethod;
(function (ConnectionOAuthHttpMethod) {
    ConnectionOAuthHttpMethod["GET"] = "GET";
    ConnectionOAuthHttpMethod["POST"] = "POST";
    ConnectionOAuthHttpMethod["PUT"] = "PUT";
})(ConnectionOAuthHttpMethod || (ConnectionOAuthHttpMethod = {}));
export var CreateConnectionOAuthRequestParameters;
(function (CreateConnectionOAuthRequestParameters) {
    CreateConnectionOAuthRequestParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateConnectionOAuthRequestParameters || (CreateConnectionOAuthRequestParameters = {}));
export var CreateConnectionAuthRequestParameters;
(function (CreateConnectionAuthRequestParameters) {
    CreateConnectionAuthRequestParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateConnectionAuthRequestParameters || (CreateConnectionAuthRequestParameters = {}));
export var CreateConnectionRequest;
(function (CreateConnectionRequest) {
    CreateConnectionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateConnectionRequest || (CreateConnectionRequest = {}));
export var ConnectionState;
(function (ConnectionState) {
    ConnectionState["AUTHORIZED"] = "AUTHORIZED";
    ConnectionState["AUTHORIZING"] = "AUTHORIZING";
    ConnectionState["CREATING"] = "CREATING";
    ConnectionState["DEAUTHORIZED"] = "DEAUTHORIZED";
    ConnectionState["DEAUTHORIZING"] = "DEAUTHORIZING";
    ConnectionState["DELETING"] = "DELETING";
    ConnectionState["UPDATING"] = "UPDATING";
})(ConnectionState || (ConnectionState = {}));
export var CreateConnectionResponse;
(function (CreateConnectionResponse) {
    CreateConnectionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateConnectionResponse || (CreateConnectionResponse = {}));
export var EndpointEventBus;
(function (EndpointEventBus) {
    EndpointEventBus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EndpointEventBus || (EndpointEventBus = {}));
export var ReplicationState;
(function (ReplicationState) {
    ReplicationState["DISABLED"] = "DISABLED";
    ReplicationState["ENABLED"] = "ENABLED";
})(ReplicationState || (ReplicationState = {}));
export var ReplicationConfig;
(function (ReplicationConfig) {
    ReplicationConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicationConfig || (ReplicationConfig = {}));
export var Primary;
(function (Primary) {
    Primary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Primary || (Primary = {}));
export var Secondary;
(function (Secondary) {
    Secondary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Secondary || (Secondary = {}));
export var FailoverConfig;
(function (FailoverConfig) {
    FailoverConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FailoverConfig || (FailoverConfig = {}));
export var RoutingConfig;
(function (RoutingConfig) {
    RoutingConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RoutingConfig || (RoutingConfig = {}));
export var CreateEndpointRequest;
(function (CreateEndpointRequest) {
    CreateEndpointRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateEndpointRequest || (CreateEndpointRequest = {}));
export var EndpointState;
(function (EndpointState) {
    EndpointState["ACTIVE"] = "ACTIVE";
    EndpointState["CREATE_FAILED"] = "CREATE_FAILED";
    EndpointState["CREATING"] = "CREATING";
    EndpointState["DELETE_FAILED"] = "DELETE_FAILED";
    EndpointState["DELETING"] = "DELETING";
    EndpointState["UPDATE_FAILED"] = "UPDATE_FAILED";
    EndpointState["UPDATING"] = "UPDATING";
})(EndpointState || (EndpointState = {}));
export var CreateEndpointResponse;
(function (CreateEndpointResponse) {
    CreateEndpointResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateEndpointResponse || (CreateEndpointResponse = {}));
export var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Tag || (Tag = {}));
export var CreateEventBusRequest;
(function (CreateEventBusRequest) {
    CreateEventBusRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateEventBusRequest || (CreateEventBusRequest = {}));
export var CreateEventBusResponse;
(function (CreateEventBusResponse) {
    CreateEventBusResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateEventBusResponse || (CreateEventBusResponse = {}));
export var CreatePartnerEventSourceRequest;
(function (CreatePartnerEventSourceRequest) {
    CreatePartnerEventSourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreatePartnerEventSourceRequest || (CreatePartnerEventSourceRequest = {}));
export var CreatePartnerEventSourceResponse;
(function (CreatePartnerEventSourceResponse) {
    CreatePartnerEventSourceResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreatePartnerEventSourceResponse || (CreatePartnerEventSourceResponse = {}));
export var DeactivateEventSourceRequest;
(function (DeactivateEventSourceRequest) {
    DeactivateEventSourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeactivateEventSourceRequest || (DeactivateEventSourceRequest = {}));
export var DeauthorizeConnectionRequest;
(function (DeauthorizeConnectionRequest) {
    DeauthorizeConnectionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeauthorizeConnectionRequest || (DeauthorizeConnectionRequest = {}));
export var DeauthorizeConnectionResponse;
(function (DeauthorizeConnectionResponse) {
    DeauthorizeConnectionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeauthorizeConnectionResponse || (DeauthorizeConnectionResponse = {}));
export var DeleteApiDestinationRequest;
(function (DeleteApiDestinationRequest) {
    DeleteApiDestinationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteApiDestinationRequest || (DeleteApiDestinationRequest = {}));
export var DeleteApiDestinationResponse;
(function (DeleteApiDestinationResponse) {
    DeleteApiDestinationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteApiDestinationResponse || (DeleteApiDestinationResponse = {}));
export var DeleteArchiveRequest;
(function (DeleteArchiveRequest) {
    DeleteArchiveRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteArchiveRequest || (DeleteArchiveRequest = {}));
export var DeleteArchiveResponse;
(function (DeleteArchiveResponse) {
    DeleteArchiveResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteArchiveResponse || (DeleteArchiveResponse = {}));
export var DeleteConnectionRequest;
(function (DeleteConnectionRequest) {
    DeleteConnectionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteConnectionRequest || (DeleteConnectionRequest = {}));
export var DeleteConnectionResponse;
(function (DeleteConnectionResponse) {
    DeleteConnectionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteConnectionResponse || (DeleteConnectionResponse = {}));
export var DeleteEndpointRequest;
(function (DeleteEndpointRequest) {
    DeleteEndpointRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteEndpointRequest || (DeleteEndpointRequest = {}));
export var DeleteEndpointResponse;
(function (DeleteEndpointResponse) {
    DeleteEndpointResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteEndpointResponse || (DeleteEndpointResponse = {}));
export var DeleteEventBusRequest;
(function (DeleteEventBusRequest) {
    DeleteEventBusRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteEventBusRequest || (DeleteEventBusRequest = {}));
export var DeletePartnerEventSourceRequest;
(function (DeletePartnerEventSourceRequest) {
    DeletePartnerEventSourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeletePartnerEventSourceRequest || (DeletePartnerEventSourceRequest = {}));
export var DeleteRuleRequest;
(function (DeleteRuleRequest) {
    DeleteRuleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteRuleRequest || (DeleteRuleRequest = {}));
var ManagedRuleException = (function (_super) {
    __extends(ManagedRuleException, _super);
    function ManagedRuleException(opts) {
        var _this = _super.call(this, __assign({ name: "ManagedRuleException", $fault: "client" }, opts)) || this;
        _this.name = "ManagedRuleException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ManagedRuleException.prototype);
        return _this;
    }
    return ManagedRuleException;
}(__BaseException));
export { ManagedRuleException };
export var DescribeApiDestinationRequest;
(function (DescribeApiDestinationRequest) {
    DescribeApiDestinationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeApiDestinationRequest || (DescribeApiDestinationRequest = {}));
export var DescribeApiDestinationResponse;
(function (DescribeApiDestinationResponse) {
    DescribeApiDestinationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeApiDestinationResponse || (DescribeApiDestinationResponse = {}));
export var DescribeArchiveRequest;
(function (DescribeArchiveRequest) {
    DescribeArchiveRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeArchiveRequest || (DescribeArchiveRequest = {}));
export var DescribeArchiveResponse;
(function (DescribeArchiveResponse) {
    DescribeArchiveResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeArchiveResponse || (DescribeArchiveResponse = {}));
export var DescribeConnectionRequest;
(function (DescribeConnectionRequest) {
    DescribeConnectionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeConnectionRequest || (DescribeConnectionRequest = {}));
export var ConnectionApiKeyAuthResponseParameters;
(function (ConnectionApiKeyAuthResponseParameters) {
    ConnectionApiKeyAuthResponseParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectionApiKeyAuthResponseParameters || (ConnectionApiKeyAuthResponseParameters = {}));
export var ConnectionBasicAuthResponseParameters;
(function (ConnectionBasicAuthResponseParameters) {
    ConnectionBasicAuthResponseParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectionBasicAuthResponseParameters || (ConnectionBasicAuthResponseParameters = {}));
export var ConnectionOAuthClientResponseParameters;
(function (ConnectionOAuthClientResponseParameters) {
    ConnectionOAuthClientResponseParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectionOAuthClientResponseParameters || (ConnectionOAuthClientResponseParameters = {}));
export var ConnectionOAuthResponseParameters;
(function (ConnectionOAuthResponseParameters) {
    ConnectionOAuthResponseParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectionOAuthResponseParameters || (ConnectionOAuthResponseParameters = {}));
export var ConnectionAuthResponseParameters;
(function (ConnectionAuthResponseParameters) {
    ConnectionAuthResponseParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectionAuthResponseParameters || (ConnectionAuthResponseParameters = {}));
export var DescribeConnectionResponse;
(function (DescribeConnectionResponse) {
    DescribeConnectionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeConnectionResponse || (DescribeConnectionResponse = {}));
export var DescribeEndpointRequest;
(function (DescribeEndpointRequest) {
    DescribeEndpointRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEndpointRequest || (DescribeEndpointRequest = {}));
export var DescribeEndpointResponse;
(function (DescribeEndpointResponse) {
    DescribeEndpointResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEndpointResponse || (DescribeEndpointResponse = {}));
export var DescribeEventBusRequest;
(function (DescribeEventBusRequest) {
    DescribeEventBusRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEventBusRequest || (DescribeEventBusRequest = {}));
export var DescribeEventBusResponse;
(function (DescribeEventBusResponse) {
    DescribeEventBusResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEventBusResponse || (DescribeEventBusResponse = {}));
export var DescribeEventSourceRequest;
(function (DescribeEventSourceRequest) {
    DescribeEventSourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEventSourceRequest || (DescribeEventSourceRequest = {}));
export var EventSourceState;
(function (EventSourceState) {
    EventSourceState["ACTIVE"] = "ACTIVE";
    EventSourceState["DELETED"] = "DELETED";
    EventSourceState["PENDING"] = "PENDING";
})(EventSourceState || (EventSourceState = {}));
export var DescribeEventSourceResponse;
(function (DescribeEventSourceResponse) {
    DescribeEventSourceResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEventSourceResponse || (DescribeEventSourceResponse = {}));
export var DescribePartnerEventSourceRequest;
(function (DescribePartnerEventSourceRequest) {
    DescribePartnerEventSourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribePartnerEventSourceRequest || (DescribePartnerEventSourceRequest = {}));
export var DescribePartnerEventSourceResponse;
(function (DescribePartnerEventSourceResponse) {
    DescribePartnerEventSourceResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribePartnerEventSourceResponse || (DescribePartnerEventSourceResponse = {}));
export var DescribeReplayRequest;
(function (DescribeReplayRequest) {
    DescribeReplayRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeReplayRequest || (DescribeReplayRequest = {}));
export var ReplayDestination;
(function (ReplayDestination) {
    ReplayDestination.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplayDestination || (ReplayDestination = {}));
export var DescribeReplayResponse;
(function (DescribeReplayResponse) {
    DescribeReplayResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeReplayResponse || (DescribeReplayResponse = {}));
export var DescribeRuleRequest;
(function (DescribeRuleRequest) {
    DescribeRuleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeRuleRequest || (DescribeRuleRequest = {}));
export var RuleState;
(function (RuleState) {
    RuleState["DISABLED"] = "DISABLED";
    RuleState["ENABLED"] = "ENABLED";
})(RuleState || (RuleState = {}));
export var DescribeRuleResponse;
(function (DescribeRuleResponse) {
    DescribeRuleResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeRuleResponse || (DescribeRuleResponse = {}));
export var DisableRuleRequest;
(function (DisableRuleRequest) {
    DisableRuleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisableRuleRequest || (DisableRuleRequest = {}));
export var EnableRuleRequest;
(function (EnableRuleRequest) {
    EnableRuleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EnableRuleRequest || (EnableRuleRequest = {}));
export var ListApiDestinationsRequest;
(function (ListApiDestinationsRequest) {
    ListApiDestinationsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListApiDestinationsRequest || (ListApiDestinationsRequest = {}));
export var ListApiDestinationsResponse;
(function (ListApiDestinationsResponse) {
    ListApiDestinationsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListApiDestinationsResponse || (ListApiDestinationsResponse = {}));
export var ListArchivesRequest;
(function (ListArchivesRequest) {
    ListArchivesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListArchivesRequest || (ListArchivesRequest = {}));
export var ListArchivesResponse;
(function (ListArchivesResponse) {
    ListArchivesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListArchivesResponse || (ListArchivesResponse = {}));
export var ListConnectionsRequest;
(function (ListConnectionsRequest) {
    ListConnectionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListConnectionsRequest || (ListConnectionsRequest = {}));
export var Connection;
(function (Connection) {
    Connection.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Connection || (Connection = {}));
export var ListConnectionsResponse;
(function (ListConnectionsResponse) {
    ListConnectionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListConnectionsResponse || (ListConnectionsResponse = {}));
export var ListEndpointsRequest;
(function (ListEndpointsRequest) {
    ListEndpointsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListEndpointsRequest || (ListEndpointsRequest = {}));
export var Endpoint;
(function (Endpoint) {
    Endpoint.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Endpoint || (Endpoint = {}));
export var ListEndpointsResponse;
(function (ListEndpointsResponse) {
    ListEndpointsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListEndpointsResponse || (ListEndpointsResponse = {}));
export var ListEventBusesRequest;
(function (ListEventBusesRequest) {
    ListEventBusesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListEventBusesRequest || (ListEventBusesRequest = {}));
export var EventBus;
(function (EventBus) {
    EventBus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EventBus || (EventBus = {}));
export var ListEventBusesResponse;
(function (ListEventBusesResponse) {
    ListEventBusesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListEventBusesResponse || (ListEventBusesResponse = {}));
export var ListEventSourcesRequest;
(function (ListEventSourcesRequest) {
    ListEventSourcesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListEventSourcesRequest || (ListEventSourcesRequest = {}));
export var EventSource;
(function (EventSource) {
    EventSource.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EventSource || (EventSource = {}));
export var ListEventSourcesResponse;
(function (ListEventSourcesResponse) {
    ListEventSourcesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListEventSourcesResponse || (ListEventSourcesResponse = {}));
export var ListPartnerEventSourceAccountsRequest;
(function (ListPartnerEventSourceAccountsRequest) {
    ListPartnerEventSourceAccountsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPartnerEventSourceAccountsRequest || (ListPartnerEventSourceAccountsRequest = {}));
export var PartnerEventSourceAccount;
(function (PartnerEventSourceAccount) {
    PartnerEventSourceAccount.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PartnerEventSourceAccount || (PartnerEventSourceAccount = {}));
export var ListPartnerEventSourceAccountsResponse;
(function (ListPartnerEventSourceAccountsResponse) {
    ListPartnerEventSourceAccountsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPartnerEventSourceAccountsResponse || (ListPartnerEventSourceAccountsResponse = {}));
export var ListPartnerEventSourcesRequest;
(function (ListPartnerEventSourcesRequest) {
    ListPartnerEventSourcesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPartnerEventSourcesRequest || (ListPartnerEventSourcesRequest = {}));
export var PartnerEventSource;
(function (PartnerEventSource) {
    PartnerEventSource.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PartnerEventSource || (PartnerEventSource = {}));
export var ListPartnerEventSourcesResponse;
(function (ListPartnerEventSourcesResponse) {
    ListPartnerEventSourcesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPartnerEventSourcesResponse || (ListPartnerEventSourcesResponse = {}));
export var ListReplaysRequest;
(function (ListReplaysRequest) {
    ListReplaysRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListReplaysRequest || (ListReplaysRequest = {}));
export var Replay;
(function (Replay) {
    Replay.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Replay || (Replay = {}));
export var ListReplaysResponse;
(function (ListReplaysResponse) {
    ListReplaysResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListReplaysResponse || (ListReplaysResponse = {}));
export var ListRuleNamesByTargetRequest;
(function (ListRuleNamesByTargetRequest) {
    ListRuleNamesByTargetRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListRuleNamesByTargetRequest || (ListRuleNamesByTargetRequest = {}));
export var ListRuleNamesByTargetResponse;
(function (ListRuleNamesByTargetResponse) {
    ListRuleNamesByTargetResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListRuleNamesByTargetResponse || (ListRuleNamesByTargetResponse = {}));
export var ListRulesRequest;
(function (ListRulesRequest) {
    ListRulesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListRulesRequest || (ListRulesRequest = {}));
export var Rule;
(function (Rule) {
    Rule.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Rule || (Rule = {}));
export var ListRulesResponse;
(function (ListRulesResponse) {
    ListRulesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListRulesResponse || (ListRulesResponse = {}));
export var ListTagsForResourceRequest;
(function (ListTagsForResourceRequest) {
    ListTagsForResourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsForResourceRequest || (ListTagsForResourceRequest = {}));
export var ListTagsForResourceResponse;
(function (ListTagsForResourceResponse) {
    ListTagsForResourceResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsForResourceResponse || (ListTagsForResourceResponse = {}));
export var ListTargetsByRuleRequest;
(function (ListTargetsByRuleRequest) {
    ListTargetsByRuleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTargetsByRuleRequest || (ListTargetsByRuleRequest = {}));
export var BatchArrayProperties;
(function (BatchArrayProperties) {
    BatchArrayProperties.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BatchArrayProperties || (BatchArrayProperties = {}));
export var BatchRetryStrategy;
(function (BatchRetryStrategy) {
    BatchRetryStrategy.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BatchRetryStrategy || (BatchRetryStrategy = {}));
export var BatchParameters;
(function (BatchParameters) {
    BatchParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BatchParameters || (BatchParameters = {}));
export var DeadLetterConfig;
(function (DeadLetterConfig) {
    DeadLetterConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeadLetterConfig || (DeadLetterConfig = {}));
export var CapacityProviderStrategyItem;
(function (CapacityProviderStrategyItem) {
    CapacityProviderStrategyItem.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CapacityProviderStrategyItem || (CapacityProviderStrategyItem = {}));
export var LaunchType;
(function (LaunchType) {
    LaunchType["EC2"] = "EC2";
    LaunchType["EXTERNAL"] = "EXTERNAL";
    LaunchType["FARGATE"] = "FARGATE";
})(LaunchType || (LaunchType = {}));
export var AwsVpcConfiguration;
(function (AwsVpcConfiguration) {
    AwsVpcConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AwsVpcConfiguration || (AwsVpcConfiguration = {}));
export var NetworkConfiguration;
(function (NetworkConfiguration) {
    NetworkConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NetworkConfiguration || (NetworkConfiguration = {}));
export var PlacementConstraintType;
(function (PlacementConstraintType) {
    PlacementConstraintType["DISTINCT_INSTANCE"] = "distinctInstance";
    PlacementConstraintType["MEMBER_OF"] = "memberOf";
})(PlacementConstraintType || (PlacementConstraintType = {}));
export var PlacementConstraint;
(function (PlacementConstraint) {
    PlacementConstraint.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PlacementConstraint || (PlacementConstraint = {}));
export var PlacementStrategyType;
(function (PlacementStrategyType) {
    PlacementStrategyType["BINPACK"] = "binpack";
    PlacementStrategyType["RANDOM"] = "random";
    PlacementStrategyType["SPREAD"] = "spread";
})(PlacementStrategyType || (PlacementStrategyType = {}));
export var PlacementStrategy;
(function (PlacementStrategy) {
    PlacementStrategy.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PlacementStrategy || (PlacementStrategy = {}));
export var PropagateTags;
(function (PropagateTags) {
    PropagateTags["TASK_DEFINITION"] = "TASK_DEFINITION";
})(PropagateTags || (PropagateTags = {}));
export var EcsParameters;
(function (EcsParameters) {
    EcsParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EcsParameters || (EcsParameters = {}));
export var HttpParameters;
(function (HttpParameters) {
    HttpParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(HttpParameters || (HttpParameters = {}));
export var InputTransformer;
(function (InputTransformer) {
    InputTransformer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InputTransformer || (InputTransformer = {}));
export var KinesisParameters;
(function (KinesisParameters) {
    KinesisParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(KinesisParameters || (KinesisParameters = {}));
export var RedshiftDataParameters;
(function (RedshiftDataParameters) {
    RedshiftDataParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RedshiftDataParameters || (RedshiftDataParameters = {}));
export var RetryPolicy;
(function (RetryPolicy) {
    RetryPolicy.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RetryPolicy || (RetryPolicy = {}));
export var RunCommandTarget;
(function (RunCommandTarget) {
    RunCommandTarget.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RunCommandTarget || (RunCommandTarget = {}));
export var RunCommandParameters;
(function (RunCommandParameters) {
    RunCommandParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RunCommandParameters || (RunCommandParameters = {}));
export var SageMakerPipelineParameter;
(function (SageMakerPipelineParameter) {
    SageMakerPipelineParameter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SageMakerPipelineParameter || (SageMakerPipelineParameter = {}));
export var SageMakerPipelineParameters;
(function (SageMakerPipelineParameters) {
    SageMakerPipelineParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SageMakerPipelineParameters || (SageMakerPipelineParameters = {}));
export var SqsParameters;
(function (SqsParameters) {
    SqsParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SqsParameters || (SqsParameters = {}));
export var Target;
(function (Target) {
    Target.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Target || (Target = {}));
export var ListTargetsByRuleResponse;
(function (ListTargetsByRuleResponse) {
    ListTargetsByRuleResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTargetsByRuleResponse || (ListTargetsByRuleResponse = {}));
export var PutEventsRequestEntry;
(function (PutEventsRequestEntry) {
    PutEventsRequestEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutEventsRequestEntry || (PutEventsRequestEntry = {}));
export var PutEventsRequest;
(function (PutEventsRequest) {
    PutEventsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutEventsRequest || (PutEventsRequest = {}));
export var PutEventsResultEntry;
(function (PutEventsResultEntry) {
    PutEventsResultEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutEventsResultEntry || (PutEventsResultEntry = {}));
export var PutEventsResponse;
(function (PutEventsResponse) {
    PutEventsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutEventsResponse || (PutEventsResponse = {}));
export var PutPartnerEventsRequestEntry;
(function (PutPartnerEventsRequestEntry) {
    PutPartnerEventsRequestEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutPartnerEventsRequestEntry || (PutPartnerEventsRequestEntry = {}));
export var PutPartnerEventsRequest;
(function (PutPartnerEventsRequest) {
    PutPartnerEventsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutPartnerEventsRequest || (PutPartnerEventsRequest = {}));
export var PutPartnerEventsResultEntry;
(function (PutPartnerEventsResultEntry) {
    PutPartnerEventsResultEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutPartnerEventsResultEntry || (PutPartnerEventsResultEntry = {}));
export var PutPartnerEventsResponse;
(function (PutPartnerEventsResponse) {
    PutPartnerEventsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutPartnerEventsResponse || (PutPartnerEventsResponse = {}));
var PolicyLengthExceededException = (function (_super) {
    __extends(PolicyLengthExceededException, _super);
    function PolicyLengthExceededException(opts) {
        var _this = _super.call(this, __assign({ name: "PolicyLengthExceededException", $fault: "client" }, opts)) || this;
        _this.name = "PolicyLengthExceededException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, PolicyLengthExceededException.prototype);
        return _this;
    }
    return PolicyLengthExceededException;
}(__BaseException));
export { PolicyLengthExceededException };
export var Condition;
(function (Condition) {
    Condition.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Condition || (Condition = {}));
export var PutPermissionRequest;
(function (PutPermissionRequest) {
    PutPermissionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutPermissionRequest || (PutPermissionRequest = {}));
export var PutRuleRequest;
(function (PutRuleRequest) {
    PutRuleRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutRuleRequest || (PutRuleRequest = {}));
export var PutRuleResponse;
(function (PutRuleResponse) {
    PutRuleResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutRuleResponse || (PutRuleResponse = {}));
export var PutTargetsRequest;
(function (PutTargetsRequest) {
    PutTargetsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutTargetsRequest || (PutTargetsRequest = {}));
export var PutTargetsResultEntry;
(function (PutTargetsResultEntry) {
    PutTargetsResultEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutTargetsResultEntry || (PutTargetsResultEntry = {}));
export var PutTargetsResponse;
(function (PutTargetsResponse) {
    PutTargetsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutTargetsResponse || (PutTargetsResponse = {}));
export var RemovePermissionRequest;
(function (RemovePermissionRequest) {
    RemovePermissionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RemovePermissionRequest || (RemovePermissionRequest = {}));
export var RemoveTargetsRequest;
(function (RemoveTargetsRequest) {
    RemoveTargetsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RemoveTargetsRequest || (RemoveTargetsRequest = {}));
export var RemoveTargetsResultEntry;
(function (RemoveTargetsResultEntry) {
    RemoveTargetsResultEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RemoveTargetsResultEntry || (RemoveTargetsResultEntry = {}));
export var RemoveTargetsResponse;
(function (RemoveTargetsResponse) {
    RemoveTargetsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RemoveTargetsResponse || (RemoveTargetsResponse = {}));
export var StartReplayRequest;
(function (StartReplayRequest) {
    StartReplayRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartReplayRequest || (StartReplayRequest = {}));
export var StartReplayResponse;
(function (StartReplayResponse) {
    StartReplayResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartReplayResponse || (StartReplayResponse = {}));
export var TagResourceRequest;
(function (TagResourceRequest) {
    TagResourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagResourceRequest || (TagResourceRequest = {}));
export var TagResourceResponse;
(function (TagResourceResponse) {
    TagResourceResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagResourceResponse || (TagResourceResponse = {}));
export var TestEventPatternRequest;
(function (TestEventPatternRequest) {
    TestEventPatternRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TestEventPatternRequest || (TestEventPatternRequest = {}));
export var TestEventPatternResponse;
(function (TestEventPatternResponse) {
    TestEventPatternResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TestEventPatternResponse || (TestEventPatternResponse = {}));
export var UntagResourceRequest;
(function (UntagResourceRequest) {
    UntagResourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagResourceRequest || (UntagResourceRequest = {}));
export var UntagResourceResponse;
(function (UntagResourceResponse) {
    UntagResourceResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagResourceResponse || (UntagResourceResponse = {}));
export var UpdateApiDestinationRequest;
(function (UpdateApiDestinationRequest) {
    UpdateApiDestinationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateApiDestinationRequest || (UpdateApiDestinationRequest = {}));
export var UpdateApiDestinationResponse;
(function (UpdateApiDestinationResponse) {
    UpdateApiDestinationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateApiDestinationResponse || (UpdateApiDestinationResponse = {}));
export var UpdateArchiveRequest;
(function (UpdateArchiveRequest) {
    UpdateArchiveRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateArchiveRequest || (UpdateArchiveRequest = {}));
export var UpdateArchiveResponse;
(function (UpdateArchiveResponse) {
    UpdateArchiveResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateArchiveResponse || (UpdateArchiveResponse = {}));
export var UpdateConnectionApiKeyAuthRequestParameters;
(function (UpdateConnectionApiKeyAuthRequestParameters) {
    UpdateConnectionApiKeyAuthRequestParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateConnectionApiKeyAuthRequestParameters || (UpdateConnectionApiKeyAuthRequestParameters = {}));
export var UpdateConnectionBasicAuthRequestParameters;
(function (UpdateConnectionBasicAuthRequestParameters) {
    UpdateConnectionBasicAuthRequestParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateConnectionBasicAuthRequestParameters || (UpdateConnectionBasicAuthRequestParameters = {}));
export var UpdateConnectionOAuthClientRequestParameters;
(function (UpdateConnectionOAuthClientRequestParameters) {
    UpdateConnectionOAuthClientRequestParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateConnectionOAuthClientRequestParameters || (UpdateConnectionOAuthClientRequestParameters = {}));
export var UpdateConnectionOAuthRequestParameters;
(function (UpdateConnectionOAuthRequestParameters) {
    UpdateConnectionOAuthRequestParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateConnectionOAuthRequestParameters || (UpdateConnectionOAuthRequestParameters = {}));
export var UpdateConnectionAuthRequestParameters;
(function (UpdateConnectionAuthRequestParameters) {
    UpdateConnectionAuthRequestParameters.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateConnectionAuthRequestParameters || (UpdateConnectionAuthRequestParameters = {}));
export var UpdateConnectionRequest;
(function (UpdateConnectionRequest) {
    UpdateConnectionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateConnectionRequest || (UpdateConnectionRequest = {}));
export var UpdateConnectionResponse;
(function (UpdateConnectionResponse) {
    UpdateConnectionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateConnectionResponse || (UpdateConnectionResponse = {}));
export var UpdateEndpointRequest;
(function (UpdateEndpointRequest) {
    UpdateEndpointRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateEndpointRequest || (UpdateEndpointRequest = {}));
export var UpdateEndpointResponse;
(function (UpdateEndpointResponse) {
    UpdateEndpointResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateEndpointResponse || (UpdateEndpointResponse = {}));
