// @aws-sdk/client-eventbridge/dist-es/commands/ActivateEventSourceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ActivateEventSourceRequest } from "../models/models_0";
import { deserializeAws_json1_1ActivateEventSourceCommand, serializeAws_json1_1ActivateEventSourceCommand, } from "../protocols/Aws_json1_1";
var ActivateEventSourceCommand = (function (_super) {
    __extends(ActivateEventSourceCommand, _super);
    function ActivateEventSourceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ActivateEventSourceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "ActivateEventSourceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ActivateEventSourceRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ActivateEventSourceCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ActivateEventSourceCommand(input, context);
    };
    ActivateEventSourceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ActivateEventSourceCommand(output, context);
    };
    return ActivateEventSourceCommand;
}($Command));
export { ActivateEventSourceCommand };
