// @aws-sdk/client-eventbridge/dist-es/commands/CancelReplayCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CancelReplayRequest, CancelReplayResponse } from "../models/models_0";
import { deserializeAws_json1_1CancelReplayCommand, serializeAws_json1_1CancelReplayCommand, } from "../protocols/Aws_json1_1";
var CancelReplayCommand = (function (_super) {
    __extends(CancelReplayCommand, _super);
    function CancelReplayCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CancelReplayCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "CancelReplayCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CancelReplayRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CancelReplayResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CancelReplayCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CancelReplayCommand(input, context);
    };
    CancelReplayCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CancelReplayCommand(output, context);
    };
    return CancelReplayCommand;
}($Command));
export { CancelReplayCommand };
