// @aws-sdk/client-eventbridge/dist-es/commands/CreateArchiveCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateArchiveRequest, CreateArchiveResponse } from "../models/models_0";
import { deserializeAws_json1_1CreateArchiveCommand, serializeAws_json1_1CreateArchiveCommand, } from "../protocols/Aws_json1_1";
var CreateArchiveCommand = (function (_super) {
    __extends(CreateArchiveCommand, _super);
    function CreateArchiveCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateArchiveCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "CreateArchiveCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateArchiveRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateArchiveResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateArchiveCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateArchiveCommand(input, context);
    };
    CreateArchiveCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateArchiveCommand(output, context);
    };
    return CreateArchiveCommand;
}($Command));
export { CreateArchiveCommand };
