// @aws-sdk/client-eventbridge/dist-es/commands/CreateConnectionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateConnectionRequest, CreateConnectionResponse } from "../models/models_0";
import { deserializeAws_json1_1CreateConnectionCommand, serializeAws_json1_1CreateConnectionCommand, } from "../protocols/Aws_json1_1";
var CreateConnectionCommand = (function (_super) {
    __extends(CreateConnectionCommand, _super);
    function CreateConnectionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateConnectionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "CreateConnectionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateConnectionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateConnectionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateConnectionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateConnectionCommand(input, context);
    };
    CreateConnectionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateConnectionCommand(output, context);
    };
    return CreateConnectionCommand;
}($Command));
export { CreateConnectionCommand };
