// @aws-sdk/client-eventbridge/dist-es/commands/DeactivateEventSourceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeactivateEventSourceRequest } from "../models/models_0";
import { deserializeAws_json1_1DeactivateEventSourceCommand, serializeAws_json1_1DeactivateEventSourceCommand, } from "../protocols/Aws_json1_1";
var DeactivateEventSourceCommand = (function (_super) {
    __extends(DeactivateEventSourceCommand, _super);
    function DeactivateEventSourceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeactivateEventSourceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "DeactivateEventSourceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeactivateEventSourceRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeactivateEventSourceCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DeactivateEventSourceCommand(input, context);
    };
    DeactivateEventSourceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DeactivateEventSourceCommand(output, context);
    };
    return DeactivateEventSourceCommand;
}($Command));
export { DeactivateEventSourceCommand };
