// @aws-sdk/client-eventbridge/dist-es/commands/DeleteArchiveCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteArchiveRequest, DeleteArchiveResponse } from "../models/models_0";
import { deserializeAws_json1_1DeleteArchiveCommand, serializeAws_json1_1DeleteArchiveCommand, } from "../protocols/Aws_json1_1";
var DeleteArchiveCommand = (function (_super) {
    __extends(DeleteArchiveCommand, _super);
    function DeleteArchiveCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteArchiveCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "DeleteArchiveCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteArchiveRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteArchiveResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteArchiveCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DeleteArchiveCommand(input, context);
    };
    DeleteArchiveCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DeleteArchiveCommand(output, context);
    };
    return DeleteArchiveCommand;
}($Command));
export { DeleteArchiveCommand };
