// @aws-sdk/client-eventbridge/dist-es/commands/DescribeApiDestinationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeApiDestinationRequest, DescribeApiDestinationResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribeApiDestinationCommand, serializeAws_json1_1DescribeApiDestinationCommand, } from "../protocols/Aws_json1_1";
var DescribeApiDestinationCommand = (function (_super) {
    __extends(DescribeApiDestinationCommand, _super);
    function DescribeApiDestinationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeApiDestinationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "DescribeApiDestinationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeApiDestinationRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeApiDestinationResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeApiDestinationCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeApiDestinationCommand(input, context);
    };
    DescribeApiDestinationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeApiDestinationCommand(output, context);
    };
    return DescribeApiDestinationCommand;
}($Command));
export { DescribeApiDestinationCommand };
