// @aws-sdk/client-eventbridge/dist-es/commands/DescribeRuleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeRuleRequest, DescribeRuleResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribeRuleCommand, serializeAws_json1_1DescribeRuleCommand, } from "../protocols/Aws_json1_1";
var DescribeRuleCommand = (function (_super) {
    __extends(DescribeRuleCommand, _super);
    function DescribeRuleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeRuleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "DescribeRuleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeRuleRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeRuleResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeRuleCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeRuleCommand(input, context);
    };
    DescribeRuleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeRuleCommand(output, context);
    };
    return DescribeRuleCommand;
}($Command));
export { DescribeRuleCommand };
