// @aws-sdk/client-eventbridge/dist-es/commands/DescribeEventBusCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeEventBusRequest, DescribeEventBusResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribeEventBusCommand, serializeAws_json1_1DescribeEventBusCommand, } from "../protocols/Aws_json1_1";
var DescribeEventBusCommand = (function (_super) {
    __extends(DescribeEventBusCommand, _super);
    function DescribeEventBusCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeEventBusCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "DescribeEventBusCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeEventBusRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeEventBusResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeEventBusCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeEventBusCommand(input, context);
    };
    DescribeEventBusCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeEventBusCommand(output, context);
    };
    return DescribeEventBusCommand;
}($Command));
export { DescribeEventBusCommand };
