// @aws-sdk/client-eventbridge/dist-es/commands/DescribeArchiveCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeArchiveRequest, DescribeArchiveResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribeArchiveCommand, serializeAws_json1_1DescribeArchiveCommand, } from "../protocols/Aws_json1_1";
var DescribeArchiveCommand = (function (_super) {
    __extends(DescribeArchiveCommand, _super);
    function DescribeArchiveCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeArchiveCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "DescribeArchiveCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeArchiveRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeArchiveResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeArchiveCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeArchiveCommand(input, context);
    };
    DescribeArchiveCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeArchiveCommand(output, context);
    };
    return DescribeArchiveCommand;
}($Command));
export { DescribeArchiveCommand };
