// @aws-sdk/client-eventbridge/dist-es/commands/DescribeReplayCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeReplayRequest, DescribeReplayResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribeReplayCommand, serializeAws_json1_1DescribeReplayCommand, } from "../protocols/Aws_json1_1";
var DescribeReplayCommand = (function (_super) {
    __extends(DescribeReplayCommand, _super);
    function DescribeReplayCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeReplayCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "DescribeReplayCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeReplayRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeReplayResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeReplayCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeReplayCommand(input, context);
    };
    DescribeReplayCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeReplayCommand(output, context);
    };
    return DescribeReplayCommand;
}($Command));
export { DescribeReplayCommand };
