// @aws-sdk/client-eventbridge/dist-es/commands/DescribeEventSourceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeEventSourceRequest, DescribeEventSourceResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribeEventSourceCommand, serializeAws_json1_1DescribeEventSourceCommand, } from "../protocols/Aws_json1_1";
var DescribeEventSourceCommand = (function (_super) {
    __extends(DescribeEventSourceCommand, _super);
    function DescribeEventSourceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeEventSourceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "DescribeEventSourceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeEventSourceRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeEventSourceResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeEventSourceCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeEventSourceCommand(input, context);
    };
    DescribeEventSourceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeEventSourceCommand(output, context);
    };
    return DescribeEventSourceCommand;
}($Command));
export { DescribeEventSourceCommand };
