// @aws-sdk/client-eventbridge/dist-es/commands/EnableRuleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { EnableRuleRequest } from "../models/models_0";
import { deserializeAws_json1_1EnableRuleCommand, serializeAws_json1_1EnableRuleCommand, } from "../protocols/Aws_json1_1";
var EnableRuleCommand = (function (_super) {
    __extends(EnableRuleCommand, _super);
    function EnableRuleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    EnableRuleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "EnableRuleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: EnableRuleRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    EnableRuleCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1EnableRuleCommand(input, context);
    };
    EnableRuleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1EnableRuleCommand(output, context);
    };
    return EnableRuleCommand;
}($Command));
export { EnableRuleCommand };
