// @aws-sdk/client-eventbridge/dist-es/commands/ListTargetsByRuleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListTargetsByRuleRequest, ListTargetsByRuleResponse } from "../models/models_0";
import { deserializeAws_json1_1ListTargetsByRuleCommand, serializeAws_json1_1ListTargetsByRuleCommand, } from "../protocols/Aws_json1_1";
var ListTargetsByRuleCommand = (function (_super) {
    __extends(ListTargetsByRuleCommand, _super);
    function ListTargetsByRuleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListTargetsByRuleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "ListTargetsByRuleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListTargetsByRuleRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListTargetsByRuleResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListTargetsByRuleCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListTargetsByRuleCommand(input, context);
    };
    ListTargetsByRuleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListTargetsByRuleCommand(output, context);
    };
    return ListTargetsByRuleCommand;
}($Command));
export { ListTargetsByRuleCommand };
