// @aws-sdk/client-eventbridge/dist-es/commands/ListPartnerEventSourceAccountsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListPartnerEventSourceAccountsRequest, ListPartnerEventSourceAccountsResponse } from "../models/models_0";
import { deserializeAws_json1_1ListPartnerEventSourceAccountsCommand, serializeAws_json1_1ListPartnerEventSourceAccountsCommand, } from "../protocols/Aws_json1_1";
var ListPartnerEventSourceAccountsCommand = (function (_super) {
    __extends(ListPartnerEventSourceAccountsCommand, _super);
    function ListPartnerEventSourceAccountsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListPartnerEventSourceAccountsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "ListPartnerEventSourceAccountsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListPartnerEventSourceAccountsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListPartnerEventSourceAccountsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListPartnerEventSourceAccountsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListPartnerEventSourceAccountsCommand(input, context);
    };
    ListPartnerEventSourceAccountsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListPartnerEventSourceAccountsCommand(output, context);
    };
    return ListPartnerEventSourceAccountsCommand;
}($Command));
export { ListPartnerEventSourceAccountsCommand };
