// @aws-sdk/client-eventbridge/dist-es/commands/ListRulesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListRulesRequest, ListRulesResponse } from "../models/models_0";
import { deserializeAws_json1_1ListRulesCommand, serializeAws_json1_1ListRulesCommand } from "../protocols/Aws_json1_1";
var ListRulesCommand = (function (_super) {
    __extends(ListRulesCommand, _super);
    function ListRulesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListRulesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "ListRulesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListRulesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListRulesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListRulesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListRulesCommand(input, context);
    };
    ListRulesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListRulesCommand(output, context);
    };
    return ListRulesCommand;
}($Command));
export { ListRulesCommand };
