// @aws-sdk/client-eventbridge/dist-es/commands/ListArchivesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListArchivesRequest, ListArchivesResponse } from "../models/models_0";
import { deserializeAws_json1_1ListArchivesCommand, serializeAws_json1_1ListArchivesCommand, } from "../protocols/Aws_json1_1";
var ListArchivesCommand = (function (_super) {
    __extends(ListArchivesCommand, _super);
    function ListArchivesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListArchivesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "ListArchivesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListArchivesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListArchivesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListArchivesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListArchivesCommand(input, context);
    };
    ListArchivesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListArchivesCommand(output, context);
    };
    return ListArchivesCommand;
}($Command));
export { ListArchivesCommand };
