// @aws-sdk/client-eventbridge/dist-es/commands/ListPartnerEventSourcesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListPartnerEventSourcesRequest, ListPartnerEventSourcesResponse } from "../models/models_0";
import { deserializeAws_json1_1ListPartnerEventSourcesCommand, serializeAws_json1_1ListPartnerEventSourcesCommand, } from "../protocols/Aws_json1_1";
var ListPartnerEventSourcesCommand = (function (_super) {
    __extends(ListPartnerEventSourcesCommand, _super);
    function ListPartnerEventSourcesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListPartnerEventSourcesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "ListPartnerEventSourcesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListPartnerEventSourcesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListPartnerEventSourcesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListPartnerEventSourcesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListPartnerEventSourcesCommand(input, context);
    };
    ListPartnerEventSourcesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListPartnerEventSourcesCommand(output, context);
    };
    return ListPartnerEventSourcesCommand;
}($Command));
export { ListPartnerEventSourcesCommand };
