// @aws-sdk/client-eventbridge/dist-es/commands/ListApiDestinationsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListApiDestinationsRequest, ListApiDestinationsResponse } from "../models/models_0";
import { deserializeAws_json1_1ListApiDestinationsCommand, serializeAws_json1_1ListApiDestinationsCommand, } from "../protocols/Aws_json1_1";
var ListApiDestinationsCommand = (function (_super) {
    __extends(ListApiDestinationsCommand, _super);
    function ListApiDestinationsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListApiDestinationsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "ListApiDestinationsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListApiDestinationsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListApiDestinationsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListApiDestinationsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListApiDestinationsCommand(input, context);
    };
    ListApiDestinationsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListApiDestinationsCommand(output, context);
    };
    return ListApiDestinationsCommand;
}($Command));
export { ListApiDestinationsCommand };
