// @aws-sdk/client-eventbridge/dist-es/commands/ListConnectionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListConnectionsRequest, ListConnectionsResponse } from "../models/models_0";
import { deserializeAws_json1_1ListConnectionsCommand, serializeAws_json1_1ListConnectionsCommand, } from "../protocols/Aws_json1_1";
var ListConnectionsCommand = (function (_super) {
    __extends(ListConnectionsCommand, _super);
    function ListConnectionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListConnectionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "ListConnectionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListConnectionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListConnectionsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListConnectionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListConnectionsCommand(input, context);
    };
    ListConnectionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListConnectionsCommand(output, context);
    };
    return ListConnectionsCommand;
}($Command));
export { ListConnectionsCommand };
