// @aws-sdk/client-eventbridge/dist-es/commands/PutPartnerEventsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PutPartnerEventsRequest, PutPartnerEventsResponse } from "../models/models_0";
import { deserializeAws_json1_1PutPartnerEventsCommand, serializeAws_json1_1PutPartnerEventsCommand, } from "../protocols/Aws_json1_1";
var PutPartnerEventsCommand = (function (_super) {
    __extends(PutPartnerEventsCommand, _super);
    function PutPartnerEventsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PutPartnerEventsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "PutPartnerEventsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutPartnerEventsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: PutPartnerEventsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutPartnerEventsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1PutPartnerEventsCommand(input, context);
    };
    PutPartnerEventsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1PutPartnerEventsCommand(output, context);
    };
    return PutPartnerEventsCommand;
}($Command));
export { PutPartnerEventsCommand };
