// @aws-sdk/client-eventbridge/dist-es/commands/PutPermissionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PutPermissionRequest } from "../models/models_0";
import { deserializeAws_json1_1PutPermissionCommand, serializeAws_json1_1PutPermissionCommand, } from "../protocols/Aws_json1_1";
var PutPermissionCommand = (function (_super) {
    __extends(PutPermissionCommand, _super);
    function PutPermissionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PutPermissionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "PutPermissionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutPermissionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutPermissionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1PutPermissionCommand(input, context);
    };
    PutPermissionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1PutPermissionCommand(output, context);
    };
    return PutPermissionCommand;
}($Command));
export { PutPermissionCommand };
