// @aws-sdk/client-eventbridge/dist-es/commands/RemoveTargetsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RemoveTargetsRequest, RemoveTargetsResponse } from "../models/models_0";
import { deserializeAws_json1_1RemoveTargetsCommand, serializeAws_json1_1RemoveTargetsCommand, } from "../protocols/Aws_json1_1";
var RemoveTargetsCommand = (function (_super) {
    __extends(RemoveTargetsCommand, _super);
    function RemoveTargetsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RemoveTargetsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "RemoveTargetsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RemoveTargetsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: RemoveTargetsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RemoveTargetsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1RemoveTargetsCommand(input, context);
    };
    RemoveTargetsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1RemoveTargetsCommand(output, context);
    };
    return RemoveTargetsCommand;
}($Command));
export { RemoveTargetsCommand };
