// @aws-sdk/client-eventbridge/dist-es/commands/UpdateConnectionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateConnectionRequest, UpdateConnectionResponse } from "../models/models_0";
import { deserializeAws_json1_1UpdateConnectionCommand, serializeAws_json1_1UpdateConnectionCommand, } from "../protocols/Aws_json1_1";
var UpdateConnectionCommand = (function (_super) {
    __extends(UpdateConnectionCommand, _super);
    function UpdateConnectionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateConnectionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EventBridgeClient";
        var commandName = "UpdateConnectionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateConnectionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateConnectionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateConnectionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateConnectionCommand(input, context);
    };
    UpdateConnectionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateConnectionCommand(output, context);
    };
    return UpdateConnectionCommand;
}($Command));
export { UpdateConnectionCommand };
